### PR TITLE
Bugfix 162

### PR DESCRIPTION
**Pull request by hiteshNEC**

### CODE REVIEW

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/QueryTerm.java [hunk 1]
```diff
@@ -776,6 +776,10 @@ private void getAttribQueryV2(StringBuilder result) throws ResponseException {
 			}
 			applyOperator(attributeFilterProperty);
 			attributeFilterProperty.append(" OR ");
+			attributeFilterProperty.append('(');
+			attributeFilterProperty.append(operant.replaceAll("\"","\'"));
+			attributeFilterProperty.append(" in (select jsonb_array_elements("+charcount+"->'https://uri.etsi.org/ngsi-ld/hasValue')->>'@value'))");
```
Comment (amitrnec) [inline Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/QueryTerm.java:781]: [Minor] please change the strings in general to the matching constants as "hasValue" and' @value'.

**Review by amitrnec — APPROVED** [review]

**Review by ScorpioBroker — CHANGES_REQUESTED** [review]
Hi,
this introduces a bad grammar sql exception on the query tests. Please check your query.

org.springframework.jdbc.BadSqlGrammarException: StatementCallback; bad SQL grammar [SELECT data_without_sysattrs as data FROM entity WHERE type = 'https://uri.etsi.org/ngsi-ld/default-context/T_Query' AND (EXISTS (SELECT FROM jsonb_array_elements(data#>'{https://uri.etsi.org/ngsi-ld/default-context/P100}') as a WHERE EXISTS (SELECT FROM jsonb_array_elements(a#>'{https://uri.etsi.org/ngsi-ld/default-context/P1_P1}') as b WHERE (b#> '{https://uri.etsi.org/ngsi-ld/hasValue,0,@value}') > '0.70'::jsonb OR (0.70 in (select jsonb_array_elements(b->'https://uri.etsi.org/ngsi-ld/hasValue')->>'@value')) OR (b#> '{@value}') > '0.70'::jsonb))) AND  1=1 LIMIT 50 OFFSET 0 ]; nested exception is org.postgresql.util.PSQLException: ERROR: operator does not exist: numeric = text
  Hint: No operator matches the given name and argument type(s). You might need to add explicit type casts.
  Position: 429

**Comment by amitrnec** [comment]
> Hi,
> this introduces a bad grammar sql exception on the query tests. Please check your query.
> 
> org.springframework.jdbc.BadSqlGrammarException: StatementCallback; bad SQL grammar [SELECT data_without_sysattrs as data FROM entity WHERE type = 'https://uri.etsi.org/ngsi-ld/default-context/T_Query' AND (EXISTS (SELECT FROM jsonb_array_elements(data#>'{https://uri.etsi.org/ngsi-ld/default-context/P100}') as a WHERE EXISTS (SELECT FROM jsonb_array_elements(a#>'{https://uri.etsi.org/ngsi-ld/default-context/P1_P1}') as b WHERE (b#> '{https://uri.etsi.org/ngsi-ld/hasValue,0,@value}') > '0.70'::jsonb OR (0.70 in (select jsonb_array_elements(b->'https://uri.etsi.org/ngsi-ld/hasValue')->>'@value')) OR (b#> '{@value}') > '0.70'::jsonb))) AND 1=1 LIMIT 50 OFFSET 0 ]; nested exception is org.postgresql.util.PSQLException: ERROR: operator does not exist: numeric = text
> Hint: No operator matches the given name and argument type(s). You might need to add explicit type casts.
> Position: 429

Done

**Review by ScorpioBroker — APPROVED** [review]
Adresses the original issue and further list handling.

**Pull request merged**